src/rust/bitbox02/src/memory.rs

```diff
@@ -42,3 +42,22 @@ pub fn bootloader_hash(out: &mut [u8; 32]) {
         bitbox02_sys::memory_bootloader_hash(out.as_mut_ptr());
     }
 }
+
+pub fn get_noise_static_private_key() -> Result<zeroize::Zeroizing<[u8; 32]>, ()> {
+    let mut out = zeroize::Zeroizing::new([0u8; 32]);
+    match unsafe { bitbox02_sys::memory_get_noise_static_private_key(out.as_mut_ptr()) } {
+        true => Ok(out),
+        false => Err(()),
+    }
+}
+
+pub fn check_noise_remote_static_pubkey(pubkey: &[u8; 32]) -> bool {
+    unsafe { bitbox02_sys::memory_check_noise_remote_static_pubkey(pubkey.as_ptr()) }
+}
+
+pub fn add_noise_remote_static_pubkey(pubkey: &[u8; 32]) -> Result<(), ()> {
```

Comment:
Would it make sense for this func to take in the array wrapped in a `Zeroizing`? Similar to what you did in https://github.com/sopium/noise-rust/pull/20/files#diff-912aec4d0b5ac65968f9c18b6d88c19c.

Reply:
This is a reference, so the caller can use (or not use) Zeroizing and it works the same way.

Since this is a pubkey, zeroizing is not needed, as it is not secret.

Reply:
Right, that was to exactly to force the caller always use Zeroizing but it is indeed a public key. Not sure what I was thinking. Ignore.